Context/Context.php

```diff
@@ -64,6 +69,22 @@ public function set(ContextKey $key, $value)
         return new static($key, $value, $this);
     }
 
+    /**
+     * @todo: Implement this on the API side
+     */
+    public function with(ImplicitContextKeyed $value): Context
```

Comment:
Given `Context` is an immutable collection I went with `with`, but given PHP doesn't have overloads I'm open to other names. I considered `withSpan` but this can also be used with baggage :/

Additionally would make it make sense to update `setValue` to `withValue` as I think that's a more common name for immutable containers.

Reply:
With seems adequate to me for now.  I agree with you that `withSpan` may get confusing alongside baggage.

`withValue` also makes sense to me for immutable containers.  

Naming these attributes is difficult.  What synonym would be applicable with `Value` in this particular instance 🤔

Reply:
What about `withContextValue`? I.e. _a value that is meant to be stored in context._ This means the API would be:

* `set` => `with`
* `setValue` => `withValue`
* `with` => `withContextValue`

Reply:
Sounds good to me!  Thanks for the clear, concise explanation.